src/main/java/de/tum/cit/aet/artemis/programming/service/localci/SharedQueueManagementService.java

```diff
@@ -323,4 +334,123 @@ public Page<BuildJob> getFilteredFinishedBuildJobs(FinishedBuildJobPageableSearc
 
         return new PageImpl<>(orderedBuildJobs, buildJobIdsPage.getPageable(), buildJobIdsPage.getTotalElements());
     }
+
+    /**
+     * Estimates how long the job will be queued for on the participation ID.
+     *
+     * @param participationId the ID of the participation for which the queue release date is estimated
+     * @return the estimated queue release date as a {@link ZonedDateTime}
+     */
+    public ZonedDateTime getBuildJobEstimatedQueueDuration(long participationId) {
```

Comment:
As the result describes the DateTime _of execution_ rather than the leftover waiting time (duration), I'd rename this method. Maybe `getBuildJobEstimatedExecution`?

Reply:
or (aligned with rest of code) `getBuildJobEstimatedStartDate`?